extensions/wikia/AdEngine/js/video/vastUrlBuilder.js

```diff
@@ -47,31 +47,17 @@ define('ext.wikia.adEngine.video.vastUrlBuilder', [
 		return aspectRatio >= 1 || !isNumeric(aspectRatio) ? adSizes.horizontal : adSizes.vertical;
 	}
 
-	function buildAdUnit(slotParams, useMegaAdUnitBuilder) {
-		var adUnitBuilder = useMegaAdUnitBuilder ? megaAdUnitBuilder : regularAdUnitBuilder;
-		return adUnitBuilder.build(slotParams.pos, slotParams.src);
-	}
-
-	/**
-	 * Creates VAST URL from given parameters.
-	 * If `options.useMegaAdUnitBuilder` is not explicitly provided, the `context.opts.megaAdUnitBuilderEnabled` is used.
-	 * @param {number} aspectRatio
-	 * @param {object} slotParams
-	 * @param {object} options
-	 * @returns {string}
-	 */
 	function build(aspectRatio, slotParams, options) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has a complexity of 12 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

Reply:
What can we do about it? Extract the pre-, mid- and post-rolls logic?